File: DemoFiles/Basic/Prime/Prime_Slow.cpp
#include <fstream>
#include <string.h>
#include <stdlib.h>
#include <assert.h>

using namespace std;

typedef  unsigned long long bignum;

void findPrimes(bignum topCandidate)
	{
	ofstream f2("output.txt");
	bignum candidate = 2;
	while(candidate <= topCandidate)
		{
		bignum trialDivisor = 2;
		int prime = 1;
		while(trialDivisor * trialDivisor <= candidate)
			{
			if(candidate % trialDivisor == 0)
				{
				prime = 0;
				break;
				}
			trialDivisor++;
			}
		if(prime) f2<<candidate<<endl;
		candidate++;
		}
		f2.close();
	}

int main(int argc, char *argv[])
	{
	bignum topCandidate = 1000;
	ifstream f("input.txt");
	f>>topCandidate;
	
	f.close();
	
	findPrimes(topCandidate);
	return 0;
	}

File: DemoFiles/Basic/Prime/Prime_Slow.cs
using System;
using System.IO;

namespace PrimeNumber
{
    class PrimeNumber
    {
		static void findPrimes(int topCandidate)
		{
			StreamWriter sw = new StreamWriter("output.txt");
			int candidate = 2;
			while(candidate <= topCandidate)
			{
				int trialDivisor = 2;
				int prime = 1;
				while(trialDivisor * trialDivisor <= candidate)
				{
					if(candidate % trialDivisor == 0)
					{
						prime = 0;
						break;
					}
					trialDivisor++;
				}
				if(prime == 1)
					sw.WriteLine(candidate);
				candidate++;
			}
			sw.Close();
		}
		static void Main(string[] args)
        {
			StreamReader sr = new StreamReader("input.txt");
			String line;
			line = sr.ReadLine();
			findPrimes(Int32.Parse(line));
			sr.Close();
		}
	}
}